core/src/main/java/jenkins/model/Jenkins.java

```diff
@@ -958,11 +958,25 @@ protected void doRun() throws Exception {
 
             updateComputerList();
 
-            {// master is online now
-                Computer c = toComputer();
-                if(c!=null)
-                    for (ComputerListener cl : ComputerListener.all())
-                        cl.onOnline(c, new LogTaskListener(LOGGER, INFO));
+            {// master is online now, it's instance must always exist
+                final Computer c = toComputer();
+                if(c != null) {
+                    for (ComputerListener cl : ComputerListener.all()) {
+                        try {
+                            cl.onOnline(c, new LogTaskListener(LOGGER, INFO));
+                        } catch (Throwable t) {
+                            if (t instanceof Error) {
+                                // We propagate Runtime errors, because they are fatal.
```

Comment:
Technically "no", because one can extend `Throwable`. So sometimes static analyzers grumble about it. Is it a realistic case? [Yes](https://github.com/search?q=org%3Ajenkinsci+%22extends+throwable%22&type=Code), even [in the Jenkins core](https://github.com/jenkinsci/jenkins/blob/2311ae8a5e7afd550b13909b4b1c4ae3338274dd/core/src/main/java/hudson/model/UpdateCenter.java#L1704)